c/driver/postgresql/postgres_type.h

```diff
@@ -0,0 +1,892 @@
+    case PG_TYPE_JSONPATH:
+      return "jsonpath_recv";
+    case PG_TYPE_LINE:
+      return "line_recv";
+    case PG_TYPE_LSEG:
+      return "lseg_recv";
+    case PG_TYPE_MACADDR:
+      return "macaddr_recv";
+    case PG_TYPE_MACADDR8:
+      return "macaddr8_recv";
+    case PG_TYPE_MULTIRANGE:
+      return "multirange_recv";
+    case PG_TYPE_NAME:
+      return "namerecv";
+    case PG_TYPE_NUMERIC:
+      return "numeric_recv";
+    case PG_TYPE_OID:
+      return "oidrecv";
+    case PG_TYPE_OIDVECTOR:
+      return "oidvectorrecv";
+    case PG_TYPE_PATH:
+      return "path_recv";
+    case PG_TYPE_PG_NODE_TREE:
+      return "pg_node_tree_recv";
+    case PG_TYPE_PG_NDISTINCT:
+      return "pg_ndistinct_recv";
+    case PG_TYPE_PG_DEPENDENCIES:
+      return "pg_dependencies_recv";
+    case PG_TYPE_PG_LSN:
+      return "pg_lsn_recv";
+    case PG_TYPE_PG_MCV_LIST:
+      return "pg_mcv_list_recv";
+    case PG_TYPE_PG_DDL_COMMAND:
+      return "pg_ddl_command_recv";
+    case PG_TYPE_PG_SNAPSHOT:
+      return "pg_snapshot_recv";
+    case PG_TYPE_POINT:
+      return "point_recv";
+    case PG_TYPE_POLY:
+      return "poly_recv";
+    case PG_TYPE_RANGE:
+      return "range_recv";
+    case PG_TYPE_RECORD:
+      return "record_recv";
+    case PG_TYPE_REGCLASS:
+      return "regclassrecv";
+    case PG_TYPE_REGCOLLATION:
+      return "regcollationrecv";
+    case PG_TYPE_REGCONFIG:
+      return "regconfigrecv";
+    case PG_TYPE_REGDICTIONARY:
+      return "regdictionaryrecv";
+    case PG_TYPE_REGNAMESPACE:
+      return "regnamespacerecv";
+    case PG_TYPE_REGOPERATOR:
+      return "regoperatorrecv";
+    case PG_TYPE_REGOPER:
+      return "regoperrecv";
+    case PG_TYPE_REGPROCEDURE:
+      return "regprocedurerecv";
+    case PG_TYPE_REGPROC:
+      return "regprocrecv";
+    case PG_TYPE_REGROLE:
+      return "regrolerecv";
+    case PG_TYPE_REGTYPE:
+      return "regtyperecv";
+    case PG_TYPE_TEXT:
+      return "textrecv";
+    case PG_TYPE_TID:
+      return "tidrecv";
+    case PG_TYPE_TIME:
+      return "time_recv";
+    case PG_TYPE_TIMESTAMP:
+      return "timestamp_recv";
+    case PG_TYPE_TIMESTAMPTZ:
+      return "timestamptz_recv";
+    case PG_TYPE_TIMETZ:
+      return "timetz_recv";
+    case PG_TYPE_TSQUERY:
+      return "tsqueryrecv";
+    case PG_TYPE_TSVECTOR:
+      return "tsvectorrecv";
+    case PG_TYPE_TXID_SNAPSHOT:
+      return "txid_snapshot_recv";
+    case PG_TYPE_UNKNOWN:
+      return "unknownrecv";
+    case PG_TYPE_UUID:
+      return "uuid_recv";
+    case PG_TYPE_VARBIT:
+      return "varbit_recv";
+    case PG_TYPE_VARCHAR:
+      return "varcharrecv";
+    case PG_TYPE_VOID:
+      return "void_recv";
+    case PG_TYPE_XID8:
+      return "xid8recv";
+    case PG_TYPE_XID:
+      return "xidrecv";
+    case PG_TYPE_XML:
+      return "xml_recv";
+    default:
+      return "";
+  }
+}
+
+static inline const char* PostgresTypname(PostgresTypeId type_id) {
```

Comment:
nit: is it useful for these to be inline in the header vs in an implementation file?